mygrad/tensor_base.py

```diff
@@ -274,11 +276,24 @@ def backward(self, grad=None):
 
         if grad is not None:
             self.grad = np.asarray(grad.data if isinstance(grad, Tensor) else grad)
+
+            if not isinstance(self.grad, (np.ndarray, Real)) or not np.issubdtype(
```

Comment:
this is duplicated from operation_base - should this be pulled out into an error-handling package that just implements this check as a function call? e.g. `is_valid_gradient()`

Reply:
Yeah, that's a good call

Reply:
Implemented `is_valid_gradient`